scipp/scipp
# Plotting variances in 2d plots

Values and std. dev images are now displayed side by side with linked axes ranges. You have to use `show_variances=True` to plot the std dev.

To test:
```Python
import scippy as sp
import numpy as np
N = 1000
M = 500
xx = np.arange(N, dtype=np.float64)
yy = np.arange(M, dtype=np.float64)
x, y = np.meshgrid(xx, yy)
b = N/20.0
c = M/2.0
r = np.sqrt(((x-c)/b)**2 + ((y-c)/b)**2)
a = np.sin(r)
d1 = sp.Dataset()
d1.set_coord(sp.Dim.X, sp.Variable([sp.Dim.X], values=xx, unit=sp.units.m))
d1.set_coord(sp.Dim.Y, sp.Variable([sp.Dim.Y], values=yy, unit=sp.units.m))
d1['mydata'] = sp.Variable([sp.Dim.Y, sp.Dim.X], values=a, variances=np.random.rand(M, N)+(x==y), unit=sp.units.counts)
sp.plot(d1, show_variances=True)
```
One should get the following:
![Screenshot at 2019-06-28 13-30-26](https://user-images.githubusercontent.com/39047984/60339415-fd048c80-99a8-11e9-877e-653d4b2f5b64.png)

To save the plot as a `HTML` file and control the size of the figure:
```
sp.plot(d1, filename="test.html", show_variances=True, figsize=(1200, 300))
```
By default, the std dev is saved to a separate file called `test_err.html`.

To specify the name of the std dev file, use
```
sp.plot(d1, filename=["test.html", "stddev.html"], show_variances=True, figsize=(1200, 300))
```

Also added

Fixes #307
